client/ayon_traypublisher/plugins/create/create_csv_ingest.py

```diff
@@ -388,6 +390,54 @@ def _resolve_repre_path(
 
         return filepath
 
+    @staticmethod
+    def _validate_parents(project_name: str, product_item: ProductItem) -> list:
+        """ Ensure parent exists for provided product_item.folder_path
+
+        Args:
+            project_name (str): The project name.
+            product_item (ProductItem): The product item to inspect.
+
+        Returns:
+            list. The parent list if any
+
+        Raise:
+            ValueError: When provided folder_path parent do not exist.
+        """
+        parent_folder_path, folder_name = product_item.folder_path.rsplit("/", 1)
+        if not parent_folder_path:
+            return []
+
+        folder_data = ayon_api.get_folder_by_path(
+            project_name,
+            parent_folder_path,
+            fields=("folderType",)
+        )
```

Comment:
Is it possible to do this that way?(I suppose `product_item.folder_path` is storing `folder_path`)
```suggestion
        parent_folder_path = product_item.folder_path
        folder_data = ayon_api.get_folder_by_path(
            project_name,
            parent_folder_path,
            fields=("folderType")
        )
```

Reply:
The recommendation above wouldn't work, because it doesn't make `fields` a `tuple` anymore.

```suggestion
        parent_folder_path = product_item.folder_path
        folder_data = ayon_api.get_folder_by_path(
            project_name,
            parent_folder_path,
            fields={"folderType"}
        )
```

However, I believe `ayon_api` internally makes `fields` a `set` anyways, so we can just pass it a set. I've updated Kayla's example here.

I'm not sure why @moonyuet is recommending that however, because we're then also querying a different folder, because we're now not querying the parent folder anymore? Which I suspect was intentionally done.

So the only change would be changing the tuple to a set, which is just cosmetics at this point:

```suggestion
        parent_folder_path, folder_name = product_item.folder_path.rsplit("/", 1)
        if not parent_folder_path:
            return []

        folder_data = ayon_api.get_folder_by_path(
            project_name,
            parent_folder_path,
            fields={"folderType"}
        )
```

Reply:
> I'm not sure why @moonyuet is recommending that however, because we're then also querying a different folder, because we're now not querying the parent folder anymore? Which I suspect was intentionally done.

I misunderstood the comment here https://github.com/ynput/ayon-unreal/blob/eedb2beec8d0af53b27f93831431956a9e8d7e9d/client/ayon_unreal/api/pipeline.py#L610-L622, so ignore me then.

Reply:
I went with @iLLiCiTiT recommendation which leverage `ayon_api.get_folders`.